store/src/java/com/zimbra/cs/mailbox/LocalMailboxCacheManager.java

```diff
@@ -0,0 +1,296 @@
+            try {
+                conn = DbPool.getConnection();
+                mailboxIds = DbMailbox.listMailboxes(conn, null);
+                cache = createCache();
+            }catch (ServiceException e) {
+                ZimbraLog.mailbox.error("creating cache manager: ", e);
+                throw new CacheManagerException("creating cache manager: ", e.getCause());
+            } finally {
+                DbPool.quietClose(conn);
+            }
+        }
+    }
+
+    @Override
+    public Collection<Integer> getMailboxIds() {
+        return mailboxIds.values();
+    }
+
+    public Set<String> getAccountIds(){
+        return mailboxIds.keySet();
+    }
+
+    @Override
+    public Set<Map.Entry<String, Integer>> getMailboxIdsByAccountId(){
+        return mailboxIds.entrySet();
+    }
+
+    @Override
+    public int getMailboxCount(){
+        return mailboxIds.size();
+    }
+
+    @Override
+    public Integer getMailboxKey(String accountId){
+        return mailboxIds.get(accountId);
+    }
+
+    @Override
+    public void removeMailboxId(String accountId){
+        mailboxIds.remove(accountId);;
+    }
+
+    public MailboxMap createCache() {
+        return new MailboxMap(LC.zimbra_mailbox_manager_hardref_cache.intValue());
+    }
+
+    @VisibleForTesting
+    public void clearCache() {
+        cache.clear();
+        mailboxIds.clear();
+    }
+
+    public Object retrieveFromCache(int mailboxId, boolean trackGC, MailboxManager mailboxManager) throws MailServiceException {
+        synchronized (mailboxManager) {
+            Object cached = cache.get(mailboxId, trackGC);
+            if (cached instanceof MailboxMaintenance) {
+                MailboxMaintenance maintenance = (MailboxMaintenance) cached;
+                if (!maintenance.canAccess()) {
+                    if (mailboxManager.isMailboxLockedOut(maintenance.getAccountId())) {
+                        throw MailServiceException.MAINTENANCE(mailboxId, "mailbox locked out for maintenance");
+                    } else {
+                        throw MailServiceException.MAINTENANCE(mailboxId);
+                    }
+                }
+                if (maintenance.getMailbox() != null) {
+                    return maintenance.getMailbox();
+                }
+            }
+            // if we've retrieved NULL or a Mailbox or an accessible lock, return it
+            return cached;
+        }
+    }
+
+    @Override
+    public int getMailboxCacheSize() {
+        return cache.size();
+    }
+
+    @Override
+    public Collection<Object> getAllLoadedMailboxes(){
+        return cache.values();
+    }
+
+    @Override
+    public Set<Map.Entry<Integer, Object>> getMailboxesById(){
+        return cache.entrySet();
+    }
+
+    @Override
+    public Object getMailbox(int mailboxId){
+        return cache.get(mailboxId);
+    }
+
+    @Override
+    public Mailbox cacheMailbox(Mailbox mailbox, MailboxManager mailboxManager) {
```

Comment:
You should remove the `mailboxManager` parameter from `MailboxCacheManager#cacheMailbox` definition.

Reply:
removed